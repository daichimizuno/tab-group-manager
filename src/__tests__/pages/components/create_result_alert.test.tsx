import { render, RenderResult } from "@testing-library/react";
import React from "react";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import CreateResultAlert from "../../../pages/components/create_result_alert";

let component: RenderResult;

beforeEach(() => {
  jest.resetAllMocks();
});

afterEach(() => {
  jest.resetAllMocks();
});

describe("CreateTabGroupComponentの試験", () => {
  describe("コンポーネントテスト", () => {
    describe("正常系", () => {
      test("スナップショットが前回の結果と一致しているかどうか", () => {
        let tree = renderer
          .create(<CreateResultAlert success={true} description="test" />)
          .toJSON() as ReactTestRendererJSON;
        expect(tree).toMatchSnapshot();
      });

      test.each([
        { success: true, description: "これはsuccssがtrueの場合" },
        { success: false, description: "これはsuccssがfalseの場合" },
      ])(
        "successが$successでdescriptionが$descriptionのとき、[$description]が表示されていること)",
        ({ success, description }) => {
          component = render(
            <CreateResultAlert success={success} description={description} />
          );
          const alert = component.getByRole("alert");

          expect(alert).toHaveTextContent(description);
        }
      );
    });
  });
});
