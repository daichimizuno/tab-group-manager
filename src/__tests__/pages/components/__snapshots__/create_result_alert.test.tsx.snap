// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTabGroupComponentの試験 コンポーネントテスト 正常系 スナップショットが前回の結果と一致しているかどうか 1`] = `
<div
  className="MuiBox-root css-1yuhvjn"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-filledSuccess MuiAlert-filled css-1brxvf4-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="SuccessOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-acap47-MuiAlert-message"
    >
      test
    </div>
  </div>
</div>
`;
